app/src/main/java/com/example/mytrainingpal/states/TodaysMusclePainEntryState.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.mytrainingpal.states
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.remember
+import com.example.mytrainingpal.model.entities.MusclePainEntry
+import com.example.mytrainingpal.model.view_models.MusclePainEntryViewModel
+import com.example.mytrainingpal.util.todayDate
+
+
+// A composable serving as a state that returns today's MusclePainEntry if there is one.
+@Composable
+fun rememberTodaysMusclePainEntryState(musclePainEntryViewModel: MusclePainEntryViewModel): MusclePainEntry? {
+    val allMusclePainEntries by musclePainEntryViewModel.allMusclePainEntries.observeAsState(listOf())
+    val todayMusclePainEntryId: MusclePainEntry? = remember(allMusclePainEntries) {
+        val today = todayDate()
+        val index = allMusclePainEntries.indexOfFirst { musclePainEntry ->
+            musclePainEntry.date.compareTo(today) == 0
+        }
+        if (index > -1) {
+            allMusclePainEntries[index]
+        } else null
+    }
+    return todayMusclePainEntryId
+}
```

Comment:
@raissacavalcanti here is a composable that only serves as a state. It uses the ViewModel to access all the musclepainentries and once they are loaded it tries to find today's muscle pain entry and return it.
All that was left to do for me was to go through all entries and find today's entry. You already listened to `allMusclePainEntries` correctly.

You can see how simple it is to use in the `Home.kt`.